src/compass/s1_cslc.py

```diff
@@ -1,25 +1,50 @@
-from compass import s1_rdr2geo, s1_geo2rdr, s1_resample_burst
+from ruamel.yaml import YAML
```

Comment:
I'm unsure about a top level function that consolidates radar and geo workflows as-is. I see the value in its simplicity. The yaml parsing is redundant and without error checking. I haven't been able to find a good workaround when the high level workflow wrapper is takes only **one** argument to cfg yaml. Taking another argument for rdr or geo mode makes yaml parsing in the file and all the text/code below irrelevant. 🤣 

I tried the following wrapper around both RunConfigs so the yaml is only read once but I am unable to come up with a way to capture the yamale validation failures because the stack traces leave out validation errors. Even if validation errors from Yamale can be captured when validation fails, the output would messy and ambiguous.

```python
import sys

import yamale

from compass.utils.geo_runconfig import GeoRunConfig
from compass.utils.runconfig import RunConfig


def cfg_load(f_yaml):
    cfg = {}
    mode = None
    e_rdr = None
    e_geo = None
    try:
        # try loading radar cfg
        cfg = RunConfig.load_from_yaml(f_yaml, 's1_cslc_rdr')
        mode = 'rdr'
    except Exception as e_rdr:
        pass

    try:
        # try loading geo cfg
        cfg = GeoRunConfig.load_from_yaml(f_yaml, 's1_cslc_geo')
        mode = 'geo'
    except Exception as e_geo:
        pass

    if mode is None:
        # print both stack traces
        raise ValueError

    return cfg, mode

if __name__ == "__main__":
    f_yaml = sys.argv[1]
    cfg = cfg_load(f_yaml)
    if cfg:
        print('winner')
    else:
        print('loser')
```

Reply:
I like the idea of having a cmd option, e.g. `-w/--workflow geo/radar`. But this does not seem doable together with `YamlArgparse()`. What I really like would be an independent, like the [`create_parser()` in `s1_geo_stack()`](https://github.com/vbrancat/COMPASS/blob/63782c0dc08dc961c33a0f62ab88c47927101729/src/compass/s1_geo_stack.py#L34). It's more flexible, and more importantly, extendable in the long term if we would like to add more options in the future. Shall we go for that?

Reply:
How do you like the following? https://github.com/yunjunz/COMPASS/pull/1

Reply:
Thank you @LiangJYu, I like it. I moved the call of YamlArgparse into the main() as the `entry_points` prefers an arg-less main().